src/Sylius/Bundle/CoreBundle/PaymentRequest/Checker/PaymentRequestIntegrityChecker.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\PaymentRequest\Checker;
+
+use Sylius\Bundle\CoreBundle\PaymentRequest\Command\PaymentRequestHashAwareInterface;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Sylius\Component\Payment\Repository\PaymentRequestRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class PaymentRequestIntegrityChecker implements PaymentRequestIntegrityCheckerInterface
+{
+    public function __construct(
+        private PaymentRequestRepositoryInterface $paymentRequestRepository,
+    ) {
+    }
+
+    public function check(PaymentRequestHashAwareInterface $command): PaymentRequestInterface
```

Comment:
I'm not convinced about naming of this service. It should check something, but it mostly asserts and provides request from command

Reply:
I agree, since it mainly retrieve a PaymentRequest it can be named "Provider".
@lchrusciel What about `PaymentRequestProvider` and divide the actual method to 2 methods : `provideFromCommand` and `provideFromHash` ?